main/reference/zoe-api/zoe.md

```diff
@@ -286,73 +286,63 @@ const { creatorFacet, publicFacet, creatorInvitation } = await E(Zoe).startInsta
 ```
 
 
-## E(Zoe).offer(invitation, proposal?, paymentKeywordRecord?, offerArgs)
-- **invitation**: **[Invitation](./zoe-data-types.md#invitation) | Promise&lt;Invitation>**
-- **proposal**: **Proposal** - Optional.
-- **paymentKeywordRecord**: **PaymentKeywordRecord** - Optional.
-- **offerArgs**: **Object**
+## E(Zoe).offer(invitation, proposal?, paymentKeywordRecord?, offerArgs?)
+- **invitation**: **[Invitation](./zoe-data-types.md#invitation) | Promise&lt;[Invitation](./zoe-data-types.md#invitation)>**
+- **proposal**: **[Proposal](/glossary/#proposal)** - Optional.
+- **paymentKeywordRecord**: **[PaymentKeywordRecord](./zoe-data-types.md#keywordrecord)** - Optional.
+- **offerArgs**: **[CopyRecord](/glossary/#passable)** - Optional.
 - Returns: **Promise&lt;[UserSeat](./user-seat.md)>**
 
-Used to make an offer to the contract that created the **Invitation** that is
-provided as the first argument.
+Used to make an offer to the contract that created the **invitation**.
 
-### Proposals and Payments
+<a name="proposals-and-payments"></a>
+### Proposals
 
-The invocation normally includes a **proposal** (the
-rules under which they want to exercise the offer) and **payments** that correspond 
-to the **give** property of the **proposal**. The payments will be escrowed by Zoe. If
-either the **proposal** or **payments** are empty, indicate this by
-omitting that argument or passing **undefined**, instead of passing an
-empty record.
-
-The optional **exit**'s value should be an **exitRule**, an object with three possible keys for
-key:value pairs:
-
-- **onDemand: null**: (Default) The offering party can cancel on demand.
-- **waived: null**: The offering party can't cancel and relies entirely on the smart contract to promptly finish their offer.
-- **afterDeadline**: The offer is automatically cancelled after a deadline, as determined 
-  by its **timer** and **deadline** properties.
-  **timer** must be a timer, and **deadline** must be a [BigInt](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt) value interpreted with respect to the timer.
-  Some example timers use Unix epoch time, while others count block height.
+**proposal** must be either `undefined` or a record with **give**, **want**, and/or **exit** properties
+respectively expressing conditions regarding what assets will be given, what must be
+received in exchange, and an exit rule defining how/when the offer can be canceled.
```

Comment:
... what is being given, what must be received to satisfy offer safety, and an exit rule ...

I'd say "is being given" because zoe requires that the proposal match the payments, and the contract won't see the offer if they don't match. (and the invitation will be burned.

"what must be received to satisfy offer safety" because the user can want any arbitrary thing (subject to `want patterns` provided by the contract), and it's a valid offer. If the contract doesn't match the want, then the offer gets its give back.

Is there a reasonable place to describe `want patterns`? Do we talk about `makeInvitation` from the contract author's perspective?

Reply:
> ... what is being given, what must be received to satisfy offer safety, and an exit rule ...
> 
> I'd say "is being given" because zoe requires that the proposal match the payments, and the contract won't see the offer if they don't match. (and the invitation will be burned.
> 
> "what must be received to satisfy offer safety" because the user can want any arbitrary thing (subject to `want patterns` provided by the contract), and it's a valid offer. If the contract doesn't match the want, then the offer gets its give back.

Done.

> Is there a reasonable place to describe `want patterns`?

I'll defer on this question, although https://docs.agoric.com/reference/zoe-api/zoe-data-types.html and https://docs.agoric.com/guides/zoe/proposal.html seem like reasonable locations for followup.

> Do we talk about `makeInvitation` from the contract author's perspective?

It's mentioned at https://docs.agoric.com/reference/zoe-api/zoe-contract-facet.html#zcf-makeinvitation-offerhandler-description-customproperties-proposalshape and https://docs.agoric.com/guides/zoe/#writing-and-installing-a-contract , but they could go further.

Reply:
We do not yet support want patterns. Looking forward to it though!